src/RelayProvider.ts

```diff
@@ -0,0 +1,278 @@
+
+        if (gasToSend !== undefined && gasToSend !== null) {
+            gasToSend = hexValue(BigInt(gasToSend));
+        }
+
+        /**
+         * When using a RelayProvider, and the gas field is not manually set, then the original provider within calculates a ridiculously high gas.
+         * In order to avoid this, we created a new field called forceGas, which is optional.
+         * If the user wants to manually set the gas, then she can put a value in the forceGas field, otherwise it will be
+         * automatically estimated (correctly) by the RelayClient.
+         * The reason why we added this new forceGas field is because at this point, by reading the gas field,
+         * we cannot differentiate between a user-entered gas value from the auto-calculated (and a very poor calculation)
+         * value that comes from the original provider
+         */
+
+        const fullEnvelopingRequest: UserDefinedEnvelopingRequest = {
+            relayData: {
+                ...envelopingRequest.relayData,
+                callForwarder: callForwarderValue,
+            }, 
+            request: {
+                ...envelopingRequest.request,
+                relayHub: relayHubValue,
+                gas: gasToSend,
+            }
+        }
+
+        const fullRequestConfig: RequestConfig = {
+            ...requestConfig,
+            onlyPreferredRelays: onlyPreferredRelaysValue
+        }
+
+        const relayingResult = await this.executeRelayTransaction(fullEnvelopingRequest, fullRequestConfig);
+
+        if(relayingResult.receipt){
+            const { relayRevertedOnRecipient, reason } = this._getRelayStatus(relayingResult.receipt);
+            const { transactionHash } = relayingResult.receipt;
+
+            if (relayRevertedOnRecipient) {
+                throw new Error(
+                        `Transaction Relayed but reverted on recipient - TxHash: ${transactionHash} , Reason: ${reason ?? 'unknown'}`
+                );
+            }
+        }
+
+        return relayingResult;
+    }
+
+    override async listAccounts() {
+        let accountsList = await this.jsonRpcProvider.listAccounts();
+
+        if (accountsList && Array.isArray(accountsList)) {
+            const { chainId } = await this.getNetwork();
+            const accountManager = new AccountManager(this, chainId);
+
+            const ephemeralAccounts =
+                accountManager.getAccounts();
+
+            accountsList = accountsList.concat(ephemeralAccounts);
+        }
+
+        return accountsList;
+    }
+
+    override send(method: string, params: Array<Record<string, unknown>>): Promise<unknown> {
+        if (this._useEnveloping(method, params)) {
+            const { requestConfig, envelopingTx } = params[0] as { requestConfig: RequestConfig, envelopingTx: UserDefinedEnvelopingRequest }
+            const { request } = envelopingTx;
+            if (method === 'eth_sendTransaction') {
+
+                if (!request.to) {
```

Comment:
Are we sure `to` can be found in the request object? According to the [previous code](https://github.com/rsksmart/rif-relay-client/blob/de33ebb63c57d6b7a94071baf480fbbc958aa790/src/RelayProvider.ts#L124), `to` was retrieved from `params[0]` directly and not from an inner property.

Reply:
that is how it used to be, yes. Now however, relay transactions need two parameters, the data itself (in the form of UserDefinedEnvelopingRequest) and the requestConfig object. This means that whoever is implementing the provider to carry out relay transactions must send an array of 
`{ requestConfig: RequestConfig, envelopingTx: UserDefinedEnvelopingRequest }` 
and set the useEnveloping flag in requestConfig to true.

Reply:
Ok, although I'm not sure this is compatible with a normal request, I got it.